src/lib/Stream.ts

```diff
@@ -8,11 +8,11 @@
 
 "use strict"; /* istanbul ignore next */ // ignores Typescript's __extend() function
 
-import { Promise, Thenable, BaseError, Deferred } from "ts-promise";
 import * as assert from "assert";
 
+import BaseError from "./BaseError";
 import { Transform, map, filter } from "./Transform";
-import { swallowErrors } from "./util";
+import {defer, Deferred, swallowErrors, track, TrackedPromise} from "./util";
```

Comment:
Please leave the spaces around import statements, to be consistent with the rest.

Reply:
ok